ui/src/lib.rs

```diff
@@ -23,24 +23,25 @@ enum Model {
 
 enum Msg {
     RouteChanged(Option<Route>),
-    StartGameMsg(page::start_game::Msg),
-    DemoMsg(page::demo::Msg),
-    LobbyMsg(page::lobby::Msg),
-    GameMsg(page::game::Msg),
+    Rendered(RenderInfo),
+    StartGame(page::start_game::Msg),
+    Demo(page::demo::Msg),
+    Lobby(page::lobby::Msg),
+    Game(page::game::Msg),
 }
 ////////////////////////////////////////////////////////////////
 // PRIVATE HELPERS //
 ////////////////////////////////////////////////////////////////
 
 impl Model {
-    pub fn get_session(&mut self) -> Session {
-        match &self {
-            Model::PageNotFound(session) => *session,
-            Model::Title(session) => *session,
-            Model::StartGame(sub_model) => sub_model.get_session(),
-            Model::Demo(session, _) => *session,
-            Model::Lobby(sub_model) => sub_model.get_session(),
-            Model::Game(sub_model) => sub_model.get_session(),
+    pub fn get_session_mut(&mut self) -> &mut Session {
```

Comment:
We need to update the `Session`, which means the `Session` we use in the update needs to be `mut`, which means this needs to get refactored.

I changed the name to end in `_mut` because maybe one day we will want another function that returns a non-mutable `Session`.